### PR TITLE
Start using `Option::is_none_or`

### LINKED ISSUE #17889: Start using `Option::is_none_or` around 1.84 or so
- remove the `IsNoneOr` trait
 - use `Option::is_none_or`

### PR DESCRIPTION
**Pull request by MoskalykA**
Closes #17889

### CODE REVIEW

**Comment by lnicola** [comment]
The method was stabilized in 1.82, so we'll need to wait for at least 10 more days (the issue proposed 1.84, but that's probably too conservative), and also bump `rust-version` in the manifest.

File: crates/hir-ty/src/consteval.rs [hunk 1]
```diff
@@ -287,7 +287,7 @@ pub(crate) fn const_eval_discriminant_variant(
     }
 
     let repr = db.enum_data(loc.parent).repr;
-    let is_signed = IsNoneOr::is_none_or(repr.and_then(|repr| repr.int), |int| int.is_signed());
+    let is_signed = Option::is_none_or(repr.and_then(|repr| repr.int), |int| int.is_signed());
```
Comment (lnicola) [inline crates/hir-ty/src/consteval.rs:290]: Let's use the method syntax (`opt.is_none_or(f)` instead of `Option::is_none_or(opt, f)`.

Comment (MoskalykA) [reply crates/hir-ty/src/consteval.rs]: yeah sorry !

**Comment by MoskalykA** [comment]
Oh, sorry, I didn't think it was up to me to change that.

**Comment by lnicola** [comment]
No worries! I'll merge this tomorrow.

**Pull request merged**